tests/bluetooth/ctrl_sw_privacy_unit/src/main.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2020 Demant
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr/types.h>
+#include <ztest.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#define CONFIG_BT_CTLR_RL_SIZE 8
+#define CONFIG_BT_CTLR_RPA_CACHE_SIZE 4
+#define CONFIG_BT_CTLR_PRIVACY 1
+#define CONFIG_BT_CTLR_SW_DEFERRED_PRIVACY 1
+#define CONFIG_BT_LOG_LEVEL 1
+
+#include "ll_sw/ull_filter.c"
+
+/*
+ * Unit test of SW deferred privacy data structure and related methods
+ * Tests the prpa_cache_add, prpa_cache_clear and prpa_cache_find functions
+ */
+
+#define BT_RAND_XX (rand() % 0xFF)
```

Comment:
Using rand() in unit-tests make test-failure hard to reproduce.
I would avoid doing this and instead put a hardcoded IRK here.
If the IRK value can cause the code to take different paths it would sense to add test cases for the different paths taken instead.

Reply:
Fair point, I'll change this